### PR TITLE
fix(pagination): bug #2025

### LINKED ISSUE #2025: [Bug Report] pagination can't work normal
### Element Plus version
1.0.2-beta.44

### OS/Browsers version
Edge  90.0.818.62 (64 bit)

### Vue version
3.0.11

### Reproduction Link
https://codepen.io/xingxiaowu/pen/PopbzbW?editors=1111

### Steps to reproduce
1. touch next button when currentPage4 is 2
2. touch “点我复原” button
3.  touch next button when currentPage4 is 2
4. touch “点我复原” button

### What is Expected?
pagination show 1
When the third step is performed, the handleCurrentChange method is triggered

### What is actually happening?
pagination show 2
When the third step is performed, the handleCurrentChange method is not triggered

### PR DESCRIPTION
**Pull request by XingXiaoWu**
Please make sure these boxes are checked before submitting your PR, thank you!
Closes #2025
* [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by XingXiaoWu** [comment]
#2025

File: packages/pagination/src/index.ts [hunk 1]
```diff
@@ -176,6 +176,7 @@ export default defineComponent({
 
     watch(() => props.currentPage, val => {
       internalCurrentPage.value = getValidCurrentPage(val)
+      lastEmittedPage.value = internalCurrentPage.value
```
Comment (ioslh) [inline packages/pagination/src/index.ts:179]: This patch is wrong, which will introduce breaking new bug.
As your code, `lastEmittedPage.value` will equals `internalCurrentPage.value`, thus [`emitChange `](https://github.com/element-plus/element-plus/blob/dev/packages/pagination/src/index.ts#L122) will never be triggered(unless size changed).
Maybe L178 & L179 should have order exchanged and run test again?

Comment (XingXiaoWu) [reply packages/pagination/src/index.ts:179]: No, it's right.
`props.currentPage change` only happy in parent change it. The pagination only change `internalCurrentPage.value`, in L125 we can find ` lastEmittedPage.value = internalCurrentPage.value`, so if parent change currentPage, `lastEmittedPage.value` need to equals `currentPage.value`, and `emitChange` can work normal.

If don't add L179, it's work not normal. Details in https://github.com/element-plus/element-plus/issues/2025, it have a case to show the bug.

Comment (ioslh) [reply packages/pagination/src/index.ts:179]: Sorry, I checked again, I think you are right. I'll delete my previous comment so as not to confuse others.

Comment (XingXiaoWu) [reply packages/pagination/src/index.ts:179]: nothing, thanks for your check

**Review by jw-foss — APPROVED** [review]

**Comment by JeremyWuuuuu** [comment]
We are ready to go!

**Pull request merged**